actors/evm/src/interpreter/instructions/ext.rs

```diff
@@ -18,12 +18,41 @@ pub fn extcodesize(
     system: &System<impl Runtime>,
 ) -> Result<(), StatusCode> {
     let addr = state.stack.pop();
-    // TODO we're fetching the entire block here just to get its size. We should instead use
+    // TODO (M2.2) we're fetching the entire block here just to get its size. We should instead use
     //  the ipld::block_stat syscall, but the Runtime nor the Blockstore expose it.
     //  Tracked in https://github.com/filecoin-project/ref-fvm/issues/867
-    let len = get_evm_bytecode_cid(system.rt, addr)
-        .and_then(|cid| get_evm_bytecode(system.rt, &cid))
-        .map(|bytecode| bytecode.len())?;
+    let len = match get_evm_bytecode_cid(system.rt, addr) {
+        Ok(CodeCid::EVM(cid)) =>
+        // TODO this is part of account abstraction hack where EOAs are Embryos
+        {
+            if cid == system.rt.get_code_cid_for_type(Type::Embryo) {
```

Comment:
So, this logic is a bit all over the place:

- We're putting the embryo CID into `CodeCid::EVM`.
- But we're putting the acccount CID into `CodeCid::Native`.
- And we have "code_cid == ...` checks in multiple places.

It seems like we either want to:

1. Just split `get_evm_bytecode_cid` into two methods.
2. Add some form of "or fallback" flag.